File: js_files/my_spaceship.js
function my_spaceship(string) {
    let cur_pos = 'up'
    let x_coord = 0
    let y_coord = 0

    for (let i = 0; i < string.length; i++) {
        if (string[i] === 'R') {
            if (cur_pos === 'up') {
                cur_pos = 'right'
            } else if (cur_pos === 'left') {
                cur_pos = 'up'
            } else if (cur_pos === 'down') {
                cur_pos = 'left'
            } else {
                cur_pos = 'down'
            }
        } else if (string[i] === 'L') {
            if (cur_pos === 'up') {
                cur_pos = 'left'
            } else if (cur_pos === 'left') {
                cur_pos = 'down'
            } else if (cur_pos === 'down') {
                cur_pos = 'right'
            } else {
                cur_pos = 'up'
            }
        } else if (string[i] === 'A') {
            if (cur_pos === 'up') {
                y_coord--
            } else if (cur_pos === 'left') {
                x_coord--
            } else if (cur_pos === 'down') {
                y_coord++
            } else {
                x_coord++
            }
        }
    }
    return `{x: ${x_coord}, y: ${y_coord}, direction: '${cur_pos}'}`
}

console.log(my_spaceship('RAARA'))

File: season1_c/quest04/ex05/my_spaceship.c
#include <stdio.h>
#include <string.h>

char* my_spaceship(char* param_1){
    // int size = sizeof(param_1)/sizeof(param_1[0]);
    // for (int i=0; i<size; i++) {
    //     printf("%c", param_1[i]);
    // }
    char cur_pos[] = "up", *ptr_cur;
    char up[] = "up", down[] = "down", right[] = "right", left[] = "left";
    char *ptr_up, *ptr_down, *ptr_right, *ptr_left;
    ptr_up = up;
    ptr_down = down;
    ptr_right = right;
    ptr_left = left;

    ptr_cur = cur_pos;
    int x_coord = 0, y_coord = 0;
    int size = sizeof(param_1)/sizeof(param_1[0]);
    for (int i=0; i < size; i++) {
        char val[1] = "R", *ptr_val;
        ptr_val = val;
        if (param_1[i] == *ptr_val) {
            if (*ptr_cur == *ptr_up) {
                *ptr_cur = *ptr_right;
            }
            //*ptr_val = "right";
            //*ptr_cur = *ptr_val;
            
        }       
    }
    int i = 0;
    while (cur_pos[i] != '\0') {
        printf("%c", cur_pos[i]);
        i++;
    }
    //printf("%c\n", *ptr_cur);

}

void main() {
    char* dir; // = "RAJ";
    dir = "RAV";
    my_spaceship(dir);
}